### DISCUSSION TITLE
Contribution - what questions should we be asking?

### DISCUSSION THREAD

**Discussion by ColinEberhardt** [finos/technical-oversight-committee#10] [General]
Hi All,

I'd like to get the ball rolling on refining the contribution process. Rather than dive into the detail, I think it makes sense to brainstorm a little. In support of this, I suggest we consider what questions we need to ask (either of the would-be contributor, or of ourselves) when a project or standard is proposed for contribution.

Please use this discussion to share your thoughts.

**Comment by ColinEberhardt** [comment]
Here are a few that come to mind immediately:
- Where does it fit in our landscape?
- Does it align with the purpose of FINOS (... to accelerate collaboration and innovation in financial services through the adoption of open source software, standards and best practices.)
- Does it solve the same problem as an existing FINOS project?
- Does it solve the same problem as an existing FINOS standard?
- Is it of general interest to member organisations?

**Reply by opoupeney** [reply]
This raises another question: do we allow projects to compete with each other?

**Reply by ColinEberhardt** [reply]
Agreed, we should be asking ourselves whether a project competes or complements - but let's have a debate about that later on when we've got a better idea of the full suite of questions we should be asking.

I don't want to go too deep just yet!

**Reply by rikoe** [reply]
- How ready is the project for contribution/adoption?
- Who will the maintainer(s) of the project be if adopted/contributed?
- Will the project be able to generate a strong enough community around it, i.e. how much interest will there be (not sure if this is quantifiable ahead of time tbh)

**Reply by jonfreedman** [reply]
I'd like to look at the future plans as well, in the early days of the Symphony Software Foundation it felt like there were a lot of small projects just dumped into the foundation, I think we should be looking for projects that are both actively maintained but also developed e.g. some roadmap

**Reply by martin-traverse** [reply]
I think this is echoing the existing points, but I'd say key considerations could include:

1. Appetite to use the product (or at least a gap in the market)
2. Appetite to support / contribute
3. Alignment with FINOS areas of focus
4. Competition with existing solutions (in FINOS or generally in the market)
5. Integration with the FINOS / OS landscape (technically, or more generally in terms of capabilities)

Then there's of course a second tier of considerations around legality, technical standards, security etc.

I agree with the point above that appetite can be difficult to predict and can change as a project
gathers steam, so we could try to think perhaps about viability instead? E.g. if the project solves
a common problem and ownership can reasonably be federated, if that model makes eccomonic / practical
sense for the particular problem being solved etc. Not a guarnatee of uptake of course but at least a
sense check!

I was thinking of this in terms of software contributions, but, perhaps they also apply to standards.

**Comment by stephengoldbaum** [comment]
Ah, I just submitted a separate issue asking that question. I'll close it.
